sdk/identity/identity/src/credentials/deviceCodeCredential.ts

```diff
@@ -233,46 +98,23 @@ export class DeviceCodeCredential implements TokenCredential {
    * @param options The options used to configure any requests this
    *                TokenCredential implementation might make.
    */
-  public async getToken(
+  getToken(
     scopes: string | string[],
     options?: GetTokenOptions
   ): Promise<AccessToken | null> {
     const { span, options: newOptions } = createSpan("DeviceCodeCredential-getToken", options);
-    try {
-      let tokenResponse: TokenResponse | null = null;
-      let scopeString = typeof scopes === "string" ? scopes : scopes.join(" ");
-      if (scopeString.indexOf("offline_access") < 0) {
-        scopeString += " offline_access";
-      }
 
-      // Try to use the refresh token first
-      if (this.lastTokenResponse && this.lastTokenResponse.refreshToken) {
-        tokenResponse = await this.identityClient.refreshAccessToken(
-          this.tenantId,
-          this.clientId,
-          scopeString,
-          this.lastTokenResponse.refreshToken,
-          undefined, // clientSecret not needed for device code auth
-          undefined,
-          newOptions
-        );
-      }
+    const scopeArray = typeof scopes === "object" ? scopes : [scopes];
 
-      if (tokenResponse === null) {
-        const deviceCodeResponse = await this.sendDeviceCodeRequest(scopeString, newOptions);
+    const deviceCodeRequest = {
+      deviceCodeCallback: this.userPromptCallback,
```

Comment:
I'm assuming our `DeviceCodePromptCallback` type has the same shape as the what's expected by `deviceCodeCallback`?

Reply:
Not quite sure what you're asking here

Reply:
I'm saying the `DeviceCodeCredential` takes `userPromptCallback: DeviceCodePromptCallback,` on construction. Here we're passing the callback directly to `acquireTokenByDeviceCode`. These callbacks have the same shape? Mostly I wanted to make sure we weren't exporting types from MSAL.

Reply:
Yup, they're different though compatible in this direction. 

Here's the MSAL version:
```ts
export declare type DeviceCodeResponse = {
    userCode: string;
    deviceCode: string;
    verificationUri: string;
    expiresIn: number;
    interval: number;
    message: string;
};
```

And here's the identity version:
```
export interface DeviceCodeInfo {
  /**
   * The device code that the user must enter into the verification page.
   */
  userCode: string;

  /**
   * The verification URI to which the user must navigate to enter the device
   * code.
   */
  verificationUri: string;

  /**
   * A message that may be shown to the user to instruct them on how to enter
   * the device code in the page specified by the verification URI.
   */
  message: string;
}
```